src/gui/AutomatableModelView.cpp

```diff
@@ -258,7 +279,227 @@ void AutomatableModelViewSlots::removeConnection()
+		newClip->setAutoResize(autoResize);
+	}
+	else
+	{
+		addSongAutomationNode();
+	}
+}
+void AutomatableModelViewSlots::updateSongNearestAutomationNode()
+{
+	std::vector<AutomationClip*> clips = AutomationClip::clipsForModel(m_amv->modelUntyped());
+	// getting the track without adding a new one if no track was found
+	AutomationTrack* track = getCurrentAutomationTrack(&clips, false);
+	// this needs to be checked because getCurrentAutomationTrack might give
+	// a nullptr if it can not find and add a track
+	if (track == nullptr) { return; }
+
+	// getting nearest node position
+	AutomationClip* nodeClip = nullptr;
+	TimePos nodePos = getNearestAutomationNode(track, &nodeClip);
+	if (nodeClip != nullptr)
+	{
+		// modifying its value
+		nodeClip->recordValue(nodePos, m_amv->modelUntyped()->getTrueValue());
+	}
+}
+void AutomatableModelViewSlots::removeSongNearestAutomationNode()
+{
+	std::vector<AutomationClip*> clips = AutomationClip::clipsForModel(m_amv->modelUntyped());
+	// getting the track without adding a new one if no track was found
+	AutomationTrack* track = getCurrentAutomationTrack(&clips, false);
+
+	// this needs to be checked because getCurrentAutomationTrack might give
+	// a nullptr if it can not find and add a track
+	if (track == nullptr) { return; }
+
+	AutomationClip* nodeClip = nullptr;
+	TimePos nodePos = getNearestAutomationNode(track, &nodeClip);
+	if (nodeClip != nullptr)
+	{
+		nodeClip->removeNode(nodePos);
+		// if there is no node left, the automationClip will be deleted
+		if (nodeClip->hasAutomation() == false)
+		{
+			delete nodeClip;
+		}
+	}
+}
+AutomationTrack* AutomatableModelViewSlots::getCurrentAutomationTrack(std::vector<AutomationClip*>* clips, bool canAddNewTrack)
+{
+	AutomationTrack* output = nullptr;
+	if (clips->size() > 0)
+	{
+		// selecting the track with the most amount of clips
+		// connected to this model
+		AutomationTrack* maxTrack = dynamic_cast<AutomationTrack*>((*clips)[0]->getTrack());
+		int maxTrackCount = 1;
+		for (size_t i = 1; i < clips->size(); i++)
+		{
+			int currentCount = 0;
+			for (size_t j = 0; j < clips->size(); j++)
+			{
+				if ((*clips)[i]->getTrack() == (*clips)[j]->getTrack())
+				{
+					currentCount++;
+				}
+			}
+			if (maxTrackCount < currentCount)
+			{
+				maxTrackCount = currentCount;
+				maxTrack = dynamic_cast<AutomationTrack*>((*clips)[i]->getTrack());;
+			}
+		}
+		output = maxTrack;
+	}
+	else if (canAddNewTrack == true)
+	{
+		// adding new track
+		output = new AutomationTrack(getGUI()->songEditor()->m_editor->model(), false);
+	}
+	return output;
+}
+AutomationClip* AutomatableModelViewSlots::getCurrentAutomationClip(AutomationTrack* track, bool canAddNewClip, bool searchAfter)
+{
+	AutomationClip* output = nullptr;
+	std::vector<Clip*> trackClips = track->getClips();
+	TimePos timePos = static_cast<TimePos>(Engine::getSong()->getPlayPos());
+	
 
+	bool tryAdding = false;
+	if (trackClips.size() > 0)
+	{
```

Comment:
You can check size = 0 and return from here itself.

Reply:
This would work:
```cpp
if (trackClips.empty()) { return nullptr; }
```

Reply:
Nope doesn't look like it (or my original suggestion) would. There seems to be something else going on down below with `tryAdding`

Reply:
tryAdding is there to avoid repeating code.
I think in this case your suggestion can work here. Only 1 line will be repeated.

Reply:
I was wrong, 4 lines will be repeated and the result will have In my opinion hidden returns.

Do you still want this change?